clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/cluster/AzureClusterProvider.groovy

```diff
@@ -0,0 +1,197 @@
+        }
+      } else {
+        cluster.loadBalancers = clusterDataEntry.relationships[AZURE_LOAD_BALANCERS.ns]?.collect { loadBalancerKey ->
+          Map parts = Keys.parse(azureCloudProvider, loadBalancerKey)
+          new AzureLoadBalancer(name: parts.name, region: parts.region, account: parts.account)
+        }
+
+        cluster.serverGroups = clusterDataEntry.relationships[AZURE_SERVER_GROUPS.ns]?.collect { serverGroupKey ->
+          Map parts = Keys.parse(azureCloudProvider, serverGroupKey)
+          new AzureServerGroupDescription(name: parts.serverGroup, region: parts.region,
+            application: parts.application, appName: parts.application)
+        }
+      }
+      cluster
+    }
+
+    clusters
+  }
+
+  private Map<String, AzureLoadBalancer> translateLoadBalancers(Collection<CacheData> loadBalancerData) {
+    loadBalancerData.collectEntries { loadBalancerEntry ->
+      Map<String, String> lbKey = Keys.parse(azureCloudProvider, loadBalancerEntry.id)
+      [(loadBalancerEntry.id) : new AzureLoadBalancer(name: lbKey.name, account: lbKey.account, region: lbKey.region)]
+    }
+  }
+
+  private Map<String, AzureServerGroupDescription> translateServerGroups(Collection<CacheData> serverGroupData) {
+    serverGroupData.collectEntries { serverGroupEntry ->
+      def serverGroup = objectMapper.convertValue(serverGroupEntry.attributes.serverGroup, AzureServerGroupDescription)
+      def instances = resolveRelationshipData(serverGroupEntry, AZURE_INSTANCES.ns)
+      serverGroup.instances = translateInstances(instances)
+      [(serverGroupEntry.id): serverGroup]
+    }
+  }
+
+  private static Set<AzureInstance> translateInstances(Collection<CacheData> instanceData) {
+    def instances = instanceData?.collect { instanceEntry ->
+        new AzureInstance(instanceEntry.attributes.instance)
+    } ?: []
+    // TODO (scotm) add health info to instances
+    instances as Set
+  }
+
+  private static Map<String, Set<AzureCluster>> mapResponse(Collection<AzureCluster> clusters) {
+    clusters.groupBy { it.accountName }.collectEntries { k, v -> [k, new HashSet(v)] }
+  }
+
+  private Collection<CacheData> resolveRelationshipDataForCollection(Collection<CacheData> sources, String relationship, CacheFilter cacheFilter = null) {
+    Collection<String> relationships = sources?.findResults { it.relationships[relationship]?: [] }?.flatten() ?: []
+    relationships ? cacheView.getAll(relationship, relationships, cacheFilter) : []
+  }
+
+  private Collection<CacheData> resolveRelationshipData(CacheData source, String relationship) {
+    resolveRelationshipData(source, relationship) { true }
+  }
+
+  private Collection<CacheData> resolveRelationshipData(CacheData source, String relationship, Closure<Boolean> relFilter) {
+    Collection<String> filteredRelationships = source.relationships[relationship]?.findAll(relFilter)
+    filteredRelationships ? cacheView.getAll(relationship, filteredRelationships) : []
+  }
+}
```

Comment:
This new file should go under cluster/view to be consistent with the rest of the resources